py/src/braintrust/wrappers/openai.py

```diff
@@ -143,12 +143,29 @@ def _guardrail_log_data(self, span: tracing.Span[tracing.GuardrailSpanData]) ->
     def _generation_log_data(self, span: tracing.Span[tracing.GenerationSpanData]) -> Dict[str, Any]:
         metrics = {}
         ttft = _maybe_timestamp_elapsed(span.ended_at, span.started_at)
+
         if ttft is not None:
             metrics["time_to_first_token"] = ttft
-        if span.span_data.usage is not None:
-            metrics["tokens"] = span.span_data.usage["total_tokens"]
-            metrics["prompt_tokens"] = span.span_data.usage["prompt_tokens"]
-            metrics["completion_tokens"] = span.span_data.usage["completion_tokens"]
+
+        usage = span.span_data.usage or {}
+        if "prompt_tokens" in usage: 
+            metrics["prompt_tokens"] = usage["prompt_tokens"]
+        elif "input_tokens" in usage:
+            metrics["prompt_tokens"] = usage["input_tokens"]
+        else:
+            metrics["prompt_tokens"] = 0
```

Comment:
i'd prefer to leave these as unspecified if they aren't specified by the upstream provider.